apps/etterlatte-saksbehandling-ui/client/src/components/behandling/soeknadsoversikt/familieforhold/barnepensjon/AnnenForelder.tsx

```diff
@@ -0,0 +1,128 @@
+import { redigerAnnenForelder } from '~shared/api/behandling'
+import { useApiCall } from '~shared/hooks/useApiCall'
+import { formaterEnumTilLesbarString } from '~utils/formatering/formatering'
+
+const AnnenForelderBorder = styled.div`
+  padding: 1.2em 1em 1em 0em;
+  display: flex;
+`
+
+const IconWrapper = styled.span`
+  width: 2.5rem;
+`
+
+const PersonInfoWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 0.5rem;
+`
+
+type Props = {
+  behandlingId: string
+  personopplysninger: Personopplysninger
+}
+
+export const AnnenForelderBoks = ({ behandlingId, personopplysninger }: Props) => {
+  const [redigerModus, setRedigerModus] = useState<boolean>(false)
+  const [redigerStatus, redigerAnnenForelderRequest] = useApiCall(redigerAnnenForelder)
+
+  const lagreAnnenForelder = (annenForelder: AnnenForelder) => {
+    redigerAnnenForelderRequest(
+      {
+        behandlingId: behandlingId,
+        annenForelder: annenForelder,
+      },
+      () => {
+        setTimeout(() => window.location.reload(), 2000)
+      }
+    )
+  }
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    control,
+  } = useForm<AnnenForelder>({
+    defaultValues: { ...personopplysninger?.annenForelder },
+  })
+
+  return (
+    <AnnenForelderBorder>
+      <IconWrapper>
+        <PersonIcon fontSize={IconSize.DEFAULT} />
+      </IconWrapper>
+      <PersonInfoWrapper>
+        <Heading size="small" level="3">
+          Annen forelder
+        </Heading>
+        {!redigerModus && (
+          <>
+            {personopplysninger.annenForelder == null ? (
+              <Button variant="secondary" onClick={() => setRedigerModus(true)}>
+                Legg til vurdering
+              </Button>
+            ) : (
+              <>
+                <BodyShort>{formaterEnumTilLesbarString(personopplysninger.annenForelder?.vurdering)}</BodyShort>
+                <BodyShort>{personopplysninger.annenForelder?.begrunnelse}</BodyShort>
+              </>
+            )}
+            {personopplysninger.annenForelder && (
+              <Button
+                type="button"
+                variant="secondary"
+                size="small"
+                icon={<PencilIcon aria-hidden />}
+                disabled={redigerModus}
+                onClick={() => setRedigerModus(true)}
+              >
+                Rediger
+              </Button>
+            )}
+          </>
+        )}
+        {redigerModus && (
+          <form onSubmit={handleSubmit(lagreAnnenForelder)}>
+            <ControlledRadioGruppe
+              name="vurdering"
+              control={control}
+              errorVedTomInput="Du må velge et svar"
+              legend=""
+              hideLegend={true}
+              radios={
+                <>
+                  <Radio size="small" value="KUN_EN_REGISTRERT_JURIDISK_FORELDER">
+                    {formaterEnumTilLesbarString('KUN_EN_REGISTRERT_JURIDISK_FORELDER')}
```

Comment:
Jeg tenker det er bedre å definere mappingen for tekstene eksplisitt, siden de uansett nå blir litt uheldige (`Ukjent forelder ikke norsk fnr` eks. kunne heller vært `Ukjent forelder (har ikke folkeregisteridentifikator)`).